pkg/api/v1alpha1/cluster.go

```diff
@@ -370,17 +373,31 @@ func validateMirrorConfig(clusterConfig *Cluster) error {
 		return errors.New("no value set for ECRMirror.Endpoint")
 	}
 	if caCert, set := os.LookupEnv(RegistryMirrorCAKey); set && len(caCert) > 0 {
-		content, err := ioutil.ReadFile(caCert)
+		_, err := ioutil.ReadFile(caCert)
 		if err != nil {
 			return fmt.Errorf("error reading the ca cert file %s: %v", caCert, err)
 		}
-		clusterConfig.Spec.RegistryMirrorConfiguration.CACertContent = string(content)
 	} else if clusterConfig.Spec.RegistryMirrorConfiguration.CACertContent == "" {
 		logger.Info(fmt.Sprintf("Warning: %s environment variable and caCertContent is not set, TLS verification will be disabled", RegistryMirrorCAKey))
 	}
 	return nil
 }
 
+func updateRegistryMirrorCA(clusterConfig *Cluster) error {
+	if clusterConfig.Spec.RegistryMirrorConfiguration == nil {
+		return nil
+	}
+	if caCert, set := os.LookupEnv(RegistryMirrorCAKey); set && len(caCert) > 0 {
```

Comment:
does this mean if user has provided the [caCertContent](https://github.com/aws/eks-anywhere/blob/214151dfd037f55436ed73d0687a97f2bd5b161e/pkg/api/v1alpha1/cluster_types.go#L53), the CLI will override it with the env var's value? I think it should be the other way where if this field's value isn't provided in the cluster spec then we default to the env var

Reply:
I based the precedence order similar to how [aws cli](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) does it. If a user specifies the env var `EKSA_REGISTRY_MIRROR_CA`, it will override the `caCertContent` in the config file. I believe capi/clusterctl does it this way as well

Reply:
@abhinavmpandey08 as per the [aws cli doc](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) `If you specify an option by using a parameter on the AWS CLI command line, it overrides any value from either the corresponding environment variable or a profile in the configuration file.` To me, a user setting this file in the clusterspec is similar to passing a parameter for a particular cli command, so it should override any value from env var. what do you think?

Reply:
I don't mind either way. What you said makes sense. I will just update the logic